ddtrace/appsec/_iast/_taint_tracking/aspects.py

```diff
@@ -1312,3 +1315,122 @@ def ospathsplitroot_aspect(*args: Any, **kwargs: Any) -> Any:
             iast_propagation_error_log(f"_aspect_ospathsplitroot. {e}")
 
     return os.path.splitroot(*args, **kwargs)  # type: ignore[attr-defined]
+
+
+def lstrip_aspect(orig_function: Optional[Callable], flag_added_args: int, *args: Any, **kwargs: Any) -> TEXT_TYPES:
+    if orig_function is not None and not isinstance(orig_function, BuiltinFunctionType):
+        if flag_added_args > 0:
+            args = args[flag_added_args:]
+        return orig_function(*args, **kwargs)
+
+    candidate_text = args[0]
+    args = args[flag_added_args:]
+
+    result = candidate_text.lstrip(*args, **kwargs)
+
+    if not isinstance(candidate_text, IAST.TEXT_TYPES):
+        return result
+
+    try:
+        _strip_lstrip_aspect(candidate_text, result)
+        return result
+    except Exception as e:
+        iast_propagation_error_log(f"lstrip_aspect. {e}")
+
+    return result
+
+
+def rstrip_aspect(orig_function: Optional[Callable], flag_added_args: int, *args: Any, **kwargs: Any) -> TEXT_TYPES:
+    if orig_function is not None and not isinstance(orig_function, BuiltinFunctionType):
+        if flag_added_args > 0:
+            args = args[flag_added_args:]
+        return orig_function(*args, **kwargs)
+
+    candidate_text = args[0]
+    args = args[flag_added_args:]
+
+    result = candidate_text.rstrip(*args, **kwargs)
+
+    if not isinstance(candidate_text, IAST.TEXT_TYPES):
+        return result
+
+    try:
+        ranges_new: List[TaintRange] = []
+        ranges_new_append = ranges_new.append
+
+        ranges = get_ranges(candidate_text)
+        len_result = len(result)
+        if len_result == len(candidate_text):
+            taint_pyobject_with_ranges(result, tuple(ranges))
+        else:
+            for taint_range in ranges:
+                if taint_range.start >= len_result:
+                    continue
+
+                new_length = min(taint_range.length, len_result)
```

Comment:
Shouldn't it be (len_result-taint_range.start) instead of just len_result ?

Reply:
yes, that is what I meant ;)

Reply:
I reopen the comment because it's not fixed.

Reply:
No, it's fixed but I don't know why github doesn't update the PR, check this commit: https://github.com/DataDog/dd-trace-py/pull/13266/commits/fb7b2e43fd8255feac1e3c80bdc876bd89802445